src/github.ts

```diff
@@ -1,452 +1,46 @@
-      >
-      defaults: RequestInterface["defaults"]
-      endpoint: EndpointInterface<{
-        url: string
-      }>
-    }
-    getByRepositoryId: {
-      (
-        params: Omit<
-          Endpoints["GET /repos/{owner}/{repo}/pulls/{pull_number}"]["parameters"],
-          "owner" | "repo"
-        >,
-      ): Promise<
-        Endpoints["GET /repos/{owner}/{repo}/pulls/{pull_number}"]["response"]
-      >
-      defaults: RequestInterface["defaults"]
-      endpoint: EndpointInterface<{
-        url: string
-      }>
-    }
-  }
-  checks: Octokit["checks"] & {
-    createCommitStatusByRepositoryId: {
-      (
-        params: Omit<
-          Endpoints["POST /repos/{owner}/{repo}/statuses/{sha}"]["parameters"],
-          "owner" | "repo"
-        > & {
-          repository_id: number
-        },
-      ): Promise<
-        Endpoints["POST /repos/{owner}/{repo}/statuses/{sha}"]["response"]
-      >
-      defaults: RequestInterface["defaults"]
-      endpoint: EndpointInterface<{
-        url: string
-      }>
-    }
-    getPullStatusByRepositoryId: {
-      (
-        params: Omit<
-          Endpoints["GET /repos/{owner}/{repo}/commits/{ref}/status"]["parameters"],
-          "owner" | "repo"
-        >,
-      ): Promise<
-        Endpoints["GET /repos/{owner}/{repo}/commits/{ref}/status"]["response"]
-      >
-      defaults: RequestInterface["defaults"]
-      endpoint: EndpointInterface<{
-        url: string
-      }>
+
+  switch (membershipStatus) {
+    case 204: {
+      return true
     }
-  }
-  orgs: Octokit["orgs"] & {
-    userMembershipByOrganizationId: {
-      (
-        params: Omit<
-          Endpoints["GET /orgs/{org}/members/{username}"]["parameters"],
-          "org"
-        > & { organization_id: number },
-      ): Promise<Endpoints["GET /orgs/{org}/members/{username}"]["response"]>
-      defaults: RequestInterface["defaults"]
-      endpoint: EndpointInterface<{
-        url: string
-      }>
+    case 302: {
+      logger.error(
+        `Github organization membership API responded with status code ${membershipStatus} which means the bot is not considered an organization member. This should never happen because the bot should already be installed in that organization. It's possible that the bot's credentials were not properly used for this request.`,
+      )
+      return false
     }
-    revokeUserMembershipByOrganizationId: {
-      (
-        params: Omit<
-          Endpoints["DELETE /orgs/{org}/members/{username}"]["parameters"],
-          "org"
-        > & { organization_id: number },
-      ): Promise<Endpoints["DELETE /orgs/{org}/members/{username}"]["response"]>
-      defaults: RequestInterface["defaults"]
-      endpoint: EndpointInterface<{
-        url: string
-      }>
+    default: {
+      logger.error(
+        `GitHub organization membership API responded with unexpected status code ${membershipStatus}`,
+      )
```

Comment:
I guess logger.error in both 302 and default cases aren't descriptive enough in terms of alerts/failure tracking. 
In such cases there should be differentiation between ERROR (4xx) and CRITICAL (5xx) severity
Do we have this in our logger?

Reply:
the default case might be an API change from GitHub which wasn't considered before, so it's not 100% indicative of the logic being broken. what's essential  is `204 == true, else false`.

the 302 case can pretty severe since an application might be useless if the permission checking isn't working properly. I don't know if it should be bumped to `logger.fatal` though. I generally only use `logger.fatal` for errors which can crash or break the application.

> should be differentiation between ERROR (4xx) and CRITICAL (5xx) severity

those status codes are given by the GitHub API, not by us

Reply:
> so it's not 100% indicative of the logic being broken.

but how you gonna pay attention to this if all (expected and unexpected) fail the same way?

> Sorry, I didn't quite get the question. What's the change you're suggesting here exactly?

I guess if you say we have `logger.fatal` that might be it

I think it makes sense to bump to fatal, as "fatal" only sounds like something going to be broken, but the goal should be to be able to distinguish the volume of different types of errors, warnings, errors, criticals (fatals)

if something, like `API change from GitHub` leads to the fact that our application became not compatible, though it didn't crash, it still will work not as expected, so to me it's same to `fatal` (or critical level). as we need to be able to catch it, and follow up to make it either compatible or fix the other way

Reply:
alright, bumped to fatal in 81c7d7423416edd861b527be37a3b54d9a0ae953